### ISSUE TITLE
Xiphox Not Working Properly

### ISSUE THREAD

**Issue by rbeltz48** [linuxmint/mint20-beta#163]
There is a problem with Xiphos in Linux Mint 20, LMDE 4, Debian 10 and even Fedora. The drop down menus in Xiphos don't work properly, making the application virtually useless. You can only choose the ones that show on the screen. One can't roll to the next set of books, chapters or verses. Bibletime works OK. The developers on GitHub state that it is a gtk and toolkit problem and not their problem. THey have refused to act on it. This problem surfaced sometime in mid-2019 around the launch of Debian 10. Also affects Fedora. See here and related issues: https://github.com/crosswire/xiphos/issues/956. Xiphos works OK in Mint 18/19. Any suggestions?

**Comment by clefebvre** [comment]
It looks like a GTK menu that's very tall.. maybe it shows up under the Cinnamon panel instead of showing above it?

**Comment by clefebvre** [comment]
I agree with the devs anyway. Although the design isn't great, the bug is definitely not theirs. It's either DE or toolkit.

**Comment by xenopeek** [comment]
Are the below highlighted drop down menus the ones you mean don't work correctly?

![image](https://user-images.githubusercontent.com/21159972/85227874-459a2e80-b3e0-11ea-9189-26ceccb06273.png)

Though indeed these menus are very high and may go off screen, I can scroll through them using the mousewheel, arrow keys or pointing mouse cursor at beginning or end of the menu. I compared it with Linux Mint 19.3 and these look and behave the same for me as on Linux Mint 20.

I tried it from Flathub and also from this PPA https://launchpad.net/~lafricain79/+archive/ubuntu/lafricaintest (from this article https://linuxsagas.digitaleagle.net/2020/05/08/xiphos-on-ubuntu-20-04/ showing also how to compile it yourself). Both work for me, if I'm looking at the correct menus. Below screenshot with the PPA version on Linux Mint 20 Cinnamon showing the verse menu with arrow icons.

(Xiphos was dropped from the Ubuntu 20.04 LTS repository. If you install it on Linux Mint 20 through Software Manager it comes from Flathub: https://flathub.org/apps/details/org.xiphos.Xiphos.)

![image](https://user-images.githubusercontent.com/21159972/85228139-edfcc280-b3e1-11ea-82ae-ed0b2f4b9879.png)

**Comment by clefebvre** [comment]
Thanks @xenopeek .

**Comment by rbeltz48** [comment]
I forgot to mention that the problem exists while using the MATE desktop. The up and down bullet arrows do not show up in Mint 20 MATE beta, LMDE 4 or Debian 10.4 when installed with Synaptic. Why was Xiphos dropped from the Ubuntu 20.04 LTS repository? I have been using it and installing it with Synaptic for the past 14 years. I will try Xenopeek's suggestion through the Software Manager and report back the results here. Thanks.

**Comment by rbeltz48** [comment]
We still have a problem. You were too quick in closing this issue. I installed Xiphos from the Software Manager with Flathub and it works OK with the Cinnamon desktop in LMDE 4. I installed MATE in LMDE 4 and tried to use Xiphos and it's a gtk mess. Totally unusable. Screenshots for both Cinnamon and MATE desktops are attached.

**Cinnamon**

![Screenshot from 2020-06-21 15-16-50](https://user-images.githubusercontent.com/17127007/85233309-53958280-b3d3-11ea-94cf-06bb31043ce3.png)

**MATE**

![Screenshot at 2020-06-21 15-09-56](https://user-images.githubusercontent.com/17127007/85233318-5e501780-b3d3-11ea-8e83-40710b9d043c.png)

**Comment by xenopeek** [comment]
I retested on Linux Mint 20 MATE. It looks the same there as on Linux Mint 20 Cinnamon and works fine.

As this project is about Linux Mint 20 beta bugs, this issue is closed correctly.

On LMDE 4 you can install Xiphos from the default repositories. I would suggest you do that instead of using the Flathub version. From your screenshot it looks like Xiphos needs some font that isn't installed on your LMDE 4 MATE conversion. For further help with this make a new topic the LMDE Software & Applications forum here: https://forums.linuxmint.com/viewforum.php?f=247. Include above screenshot to give others there a good clue as to what the issue is.

**Comment by rbeltz48** [comment]
Vince & Clem,

I installed Xiphos from Flathub/Flatpak as you suggested and it runs but not the way it should. Every time I open Xiphos it shows a default view and not the view I had on the screen when I closed the application. It shows Romans 8:28 along with Preview, Commentary and Dictionary views. In earlier versions of Xiphos the view stayed exactly the way I left it when I closed Xiphos. 

Also the changes I make in Preferences are not being applied after I close the application. It goes back to the default Preferences when I reopen the application. Then I have to change the Preferences to what I want every single time.

This has been rather disappointing since I had to change these things each and every time I opened Xiphos with the Flathub version and Linux Mint 20.04 with the MATE desktop. I reported this to one of the Xiphos developers and he stated the following:

From Karl Kleinpaste, Xiphos developer
_[a]_ we have observed this problem with flatpak builds of xiphos before. there is nothing we can do about it, it is not from us, and we do not know how whatever packager produced it in a manner that prevents prefs from doing the right thing.
[b] there is a PPA for xiphos exactly because of the problem of ubuntu having abandoned a package needed for xiphos' editor: https://launchpad.net/~pkgcrosswire/+archive/ubuntu/ppa
Leave behind the flatpak and install the real deal._

I followed his advice and populated Synaptic with what is needed for installing and running Xiphos correctly. Having a correct running copy of Xiphos and Bibletime is a deal breaker for me as I an an online Bible teacher and sermon presenter. 

I am telling you this so that you will know that Flathub/Flatpak is not always the best solution for third party applications running under Ubuntu/Linux Mint which did not work in the case of Xiphos. Thank you for your time.
